pgjdbc/src/main/java/org/postgresql/ssl/MakeSSL.java

```diff
@@ -62,7 +64,11 @@ private static void verifyPeerName(PGStream stream, Properties info, SSLSocket n
     if (sslhostnameverifier == null) {
       hvn = PGjdbcHostnameVerifier.INSTANCE;
       sslhostnameverifier = "PgjdbcHostnameVerifier";
-    } else {
+    }
+    else if (sslhostnameverifier.equalsIgnoreCase("com.yugabyte.ysql.YBManagedHostnameVerifier")){
+      hvn = new YBManagedHostnameVerifier(info,stream);
```

Comment:
@Sfurti-yb can we keep hvn itself as a singleton object? Construct it once and then reuse it? or creating multiple instances are required. I think one verifier object per connection is getting created. Is it right? Do we really need it? Can we have one instance of hvn created or make the HostnameVerifier itself a singleton class? cc @ashetkar

Reply:
We will have to make hvn variable static, and put it in a synchronised block. This can affect the performance during concurrent connecetions.